source/main_cr.f90

```diff
@@ -1393,77 +1393,83 @@ program maincr
   ! Dump the final state of the particle arrays
   call part_writeState(1)
 
-#ifndef FLUKA
-#ifndef STF
+#ifdef FLUKA
+
+  ! A.Mereghetti, for the FLUKA Team
+  ! last modified: 28-05-2014
+  ! collect a couple of goto statements, sending code flow
+  !   to different plotting points, which are not actually
+  !   inserted
+  490 continue
+  520 continue
+    call fluka_close
+
+#else
+
   iposc = 0
   if(ipos == 1) then ! Variable IPOS=1 -> postprocessing block present in fort.3
-    ! Open fort.10 unless already opened for BOINC
-    call f_open(unit=10, file="fort.10", formatted=.true., mode="rw",err=fErr,recl=8195)
-    call f_open(unit=110,file="fort.110",formatted=.false.,mode="w", err=fErr)
+    call f_requestUnit(fort10, unit10)
+    call f_requestUnit(fort110,unit110)
+    call f_open(unit=unit10, file=fort10, formatted=.true., mode="rw",err=fErr,status="replace",recl=8195)
+    call f_open(unit=unit110,file=fort110,formatted=.false.,mode="rw",err=fErr,status="replace")
     do ia=1,napxo,2
-      ia2=(ia+1)/2
-      iposc=iposc+1
-#ifndef CR
-      call postpr(91-ia2) ! Postprocess file "fort.(91-ia2)"
-#else
-      write(crlog,"(2(a,i0))") "SIXTRACR> Calling POSTPR Unit = ",(91-ia2),", nnuml = ",nnuml
+      iposc = iposc+1
+#ifdef STF
+#ifdef CR
+      call postpr(ia,nnuml)
```

Comment:
Why is now the logging call below the postpr? If postpr crashes etc., it won't tell you as much about where the crash was (and if it didn't crash, it now lies to you ("call*ing*) about what you are currently are waiting for.

Same issue is repeated a few places below.

Reply:
Got reshuffled when everything was moved around back and forth a few times. Will fix this.